src/HealthGPS/mapping.h

```diff
@@ -51,20 +51,8 @@ class MappingEntry {
     /// @brief Initialises a new instance of the MappingEntry class
     /// @param name Risk factor name
     /// @param level The hierarchical level
-    /// @param entity_key The associated Person property, if exists
     /// @param range The factor range
-    MappingEntry(std::string name, int level, core::Identifier entity_key, FactorRange range);
-
-    /// @brief Initialises a new instance of the MappingEntry class
-    /// @param name Risk factor name
-    /// @param level The hierarchical level
-    /// @param entity_key The associated Person property, if exists
-    MappingEntry(std::string name, int level, core::Identifier entity_key);
-
-    /// @brief Initialises a new instance of the MappingEntry class
-    /// @param name Risk factor name
-    /// @param level The hierarchical level
-    MappingEntry(std::string name, int level);
+    MappingEntry(std::string name, int level, FactorRange range = {});
```

Comment:
I'm guessing `FactorRange` defaults to `(0, 0)`. Is that safe or would it be better to use a `std::optional` for it, assuming that `(0, 0)` just indicates that there is no range specified?

This is possibly out of scope for this PR.

Reply:
It defaults to the 'empty' range, which is practically identical to an optional. I might as well just swap it out for optional now.

Reply:
Sounds good. It's nicer to have the extra type safety you get with `std::optional`.